crates/starknet-devnet/src/main.rs

```diff
@@ -198,20 +204,51 @@ async fn main() -> Result<(), anyhow::Error> {
 
     info!("Starknet Devnet listening on {}", address);
 
-    if starknet_config.dump_on == Some(DumpOn::Exit) {
-        server.with_graceful_shutdown(shutdown_signal(api.clone())).await?
-    } else {
-        server.await?
+    let empty_handle: task::JoinHandle<()> = task::spawn(async {});
+    let mut block_interval_handle = empty_handle;
+    if let BlockGeneration::Interval(timestamp) = starknet_config.block_generation_on {
+        // use JoinHandle to run block interval creation as a task
+        block_interval_handle = task::spawn(create_block_interval(api.clone(), timestamp));
     }
 
+    // run server also as a JoinHandle
+    let server_handle =
+        task::spawn(server.with_graceful_shutdown(shutdown_signal(api.clone())).into_future());
```

Comment:
Isn't this SIGINT handler installed twice if block generation is in interval mode?

Reply:
Yes! And this is what we need to avoid hanging on of another task (both need to handle ctrl c).

Reply:
I tried to use tokio::sync::broadcast to handle ctrl c but without success. Chat GPT suggested tokio::join and that worked!